Analysis/PhotonEnergyCorrect/include/photonCorrectionProcessor.h

```diff
@@ -0,0 +1,102 @@
+#ifndef photonCorrectionProcessor_h
+#define photonCorrectionProcessor_h 1
+#include "marlin/Processor.h"
+
+#include "photonCorrector.h"
+#include "TFile.h"
+#include "TH2F.h"
+
+//using namespace marlin ;
+
+class photonCorrectionProcessor : public marlin::Processor {
+  
+ public:
+  
+  virtual Processor*  newProcessor() { return new photonCorrectionProcessor ; }
+  
+  photonCorrectionProcessor() ;
+  
+  /** Called at the begin of the job before anything is read.
+   * Use to initialize the processor, e.g. book histograms.
+   */
+  virtual void init() ;
+  
+  /** Called for every run.
+   */
+  virtual void processRunHeader( LCRunHeader* run ) ;
+    
+  /** Called for every event - the working horse.
+   */
+  virtual void processEvent( LCEvent * evt ) ; 
+
+  virtual void check( LCEvent * evt ) ; 
+  
+  /** Called after data processing for clean up.
+   */
+  virtual void end() ;
+
+  photonCorrectionProcessor(const photonCorrectionProcessor&) = delete;
+  photonCorrectionProcessor& operator=(const photonCorrectionProcessor&) = delete;
+
+ protected:
+
+  photonCorrector* _photonCorrector{};
+
+  bool _modifyPFOenergies{};
+  bool _validationPlots{};
+  bool _interModGaps_haveBeenCorrectedAtHitLevel{};
+
+  std::string _validationPlotFilename{};
+  TFile* _fout{};
```

Comment:
Can you use the AIDA facility instead ? Easier for validation plots

Reply:
done